lib/stages/frbNetworkProcess.cpp

```diff
@@ -298,5 +228,335 @@ void frbNetworkProcess::main_thread() {
+    parse_chime_host_name(rack, node, nos, my_node_id);
+    for (int i = 0; i < number_of_subnets; i++) {
+        // construct an local address for each of the four VLANs 10.6.0.0/16..10.9.0.0/16
+        std::string ip_addr = fmt::format("10.{:d}.{:d}.1{:d}", i + 6, nos + rack, node);
+        DEBUG("{} ", ip_addr);
+
+        // parse the local address and port into a `sockaddr` struct
+        struct sockaddr_in addr;
+        std::memset((char*)&addr, 0, sizeof(addr));
+        addr.sin_family = AF_INET;
+        inet_pton(AF_INET, ip_addr.c_str(), &addr.sin_addr);
+        addr.sin_port = htons(udp_frb_port_number);
+
+        // bind a sending UDP socket to the local address and port
+        int sock_fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+        if (sock_fd < 0) {
+            FATAL_ERROR("Network Thread: socket() failed: {:s} ", strerror(errno));
+            return -1;
+        }
+        if (bind(sock_fd, (struct sockaddr*)&addr, sizeof(addr)) < 0) {
+            FATAL_ERROR("port binding failed ");
+            return -1;
+        }
+
+        // use a larger send buffer for the socket
+        const int n = 256 * 1024 * 1024;
+        if (setsockopt(sock_fd, SOL_SOCKET, SO_SNDBUF, (void*)&n, sizeof(n)) < 0) {
+            FATAL_ERROR("Network Thread: setsockopt() failed: %s ", strerror(errno));
+            return -1;
+        }
+
+        src_sockets.push_back({addr, sock_fd});
+    }
+
+    /* every node is introducing packets to the network. To achive load balancing a sequece_id is
+       computed for each node this will make sure none of the catalyst switches are overloaded with
+       traffic at a given point of time ofcourse this will be usefull when chrony is sufficuently
+       synchronized across all nodes..
+    */
+
+    int my_sequence_id =
+        (int)(my_node_id / 128) + 2 * ((my_node_id % 128) / 8) + 32 * (my_node_id % 8);
+    return my_sequence_id;
+}
+
+
+int frbNetworkProcess::initialize_destinations() {
+    // reading the L1 ip addresses from the config file
+    const std::vector<std::string> link_ip =
+        config.get<std::vector<std::string>>(unique_name, "L1_node_ips");
+    for (size_t i = 0; i < link_ip.size(); i++) {
+        sockaddr_in addr;
+        memset(&addr, 0, sizeof(addr));
+        addr.sin_family = AF_INET;
+        inet_pton(AF_INET, link_ip[i].c_str(), &addr.sin_addr);
+        if (!dest_sockets.count(addr.sin_addr.s_addr)) {
+            // new destination, initialize the entry in `dest_sockets`
+            addr.sin_port = htons(udp_frb_port_number);
+            int sending_socket = get_vlan_from_ip(link_ip[i].c_str()) - 6;
+            dest_sockets.insert(
+                {addr.sin_addr.s_addr, DestIpSocket{link_ip[i], addr, sending_socket}});
+        }
+        stream_dest.push_back(std::ref(dest_sockets.at(addr.sin_addr.s_addr)));
+    }
+    return link_ip.size();
+}
+
+void frbNetworkProcess::set_destination_active_callback(connectionInstance& conn,
```

Comment:
Hmm, I don't have enough familiarity with this, and haven't done anything with Coco since it's been adopted. Should we have a call for you to explain to me how this should work?

I'm also OK with taking out this endpoint in this round, and working on a proper implementation for the next cycle. It was an afterthought that I only added at the end, and the only reason for it was as an to avoid pinging nodes that we know are not on.

Reply:
I'd baseline for removing this endpoint for now since things are busy at the moment getting everything ready.  But I'm happy to have a call, let's talk on slack to coordinate.

Reply:
Resolved by removing the endpoint and making an empty string entry in `L1_node_ips` deactivate sending that beam out to a destination.